R/tree_accessors.R

```diff
@@ -2307,6 +2364,305 @@ setMethod(
   }
 )
 
+match_path_by_pos <- function(kidlst, path) {
+  ret <- -1
+  nmval_pairs <- lapply(
+    kidlst,
+    function(kd) {
+      pos <- tree_pos(kd)
+      c(obj_name(tail(pos_splits(pos), 1)[[1]]),
+        rawvalues(tail(pos_splvals(pos), 1))[[1]])
+    }
+  )
+
+  matches <- vapply(
+    nmval_pairs,
+    function(pair) {
+      (pair[1] == path[1]) && (path[2] %in% c(pair[2], "*"))
+    },
+    TRUE
+  )
+  if (any(matches))
+    ret <- which(matches)
+  ret
+
+}
+
+## close to a duplicate of tt_at_path, but... not quite :(
+#' @rdname int_methods
+coltree_at_path <- function(obj, path, ...) {
+  if (length(path) == 0)
+    return(obj)
+  stopifnot(
+    is(path, "character"),
+    length(path) > 0,
+    !anyNA(path)
+  )
+  if (any(grepl("@content", path, fixed = TRUE)))
+    stop("@content token is not valid for column paths.")
+
+
+  ## if(obj_name(obj) == path[1]) {
+  ##   path <- path[-1]
+  ## }
+
+  cur <- obj
+  curpath <- path
+  while (length(curpath) > 0) {
+    kids <- tree_children(cur)
+    kidmatch <- match_path_by_pos(kids, curpath)
+    curname <- curpath[1]
+    if (curname == "@content")
+      stop("@content is not a valid token for a column path")
+    else if (kidmatch > 0)
+      cur <- kids[[kidmatch]]
+    else
+      stop("Path appears invalid for this tree at step ", curname)
+    curpath <- curpath[-(1:2)]  # name and value, both consumed due to structure
+  }
+  cur
+}
+
+`coltree_at_path<-` <- function(obj, path, value) {
+  obj <- coltree(obj) ## noop if it already is
+  if (any(grepl("@content", path, fixed = TRUE)))
+    stop("@content token is not valid for column paths.")
+  ## we don't have intermediary structures in the coltree model
+  ## **after the first one** Yes this is bad and wrong but
+
+  ## its how it is.
+  ## ie it goes straight from A: Drug X -> F
+  ## in the split_cols_by("ARM") %>% split_cols_by("SEX") case
+  ## but we want to use the do_recursive_replace machinery
+  ## that is already well tested
+  trimmed_path <- path[c(1, seq(2, length(path), by = 2))]
+
+  do_recursive_replace(obj, trimmed_path, value = value)
+}
+
+
+#' Set visibility of column counts for a group of sibling facets
+#'
+#' @inheritParams gen_args
+#' @param path character. The path *to the parent of the
+#' desired siblings*. The last element in the path should
+#' be a split name.
+#' @return obj, modified with the desired columncount
+#' display behavior
+`facet_colcounts_visible<-` <- function(obj, path, value) {
+  coldf <- make_col_df(obj, visible_only = FALSE)
+  allpaths <- coldf$path
+  lenpath <- length(path)
+  match_paths <- vapply(allpaths, function(path_i) {
+    (length(path_i) == lenpath + 1) &&
+      (all(head(path_i, -1) == path))
+  }, TRUE)
+  for (curpath in allpaths[match_paths]) {
+    colcount_visible(obj, curpath) <- value
+  }
+  obj
+}
+
+#' Get or set column count for a facet in column space
+#'
+#' @inheritParams gen_args
+#' @param path character. This path must end on a
```

Comment:
```suggestion
#' @param path (`string`)\cr this path must end on a
```

Reply:
This is incorrect. A string, in as much as it exists at all (which, again,  in R it doesnt), has length one, whereas a path is a character vector of any length.

Reply:
I will admit I forgot to put the weird new stuff around it though...